engine/src/main/java/org/terasology/entitySystem/systems/RenderSystem.java

```diff
@@ -16,16 +16,32 @@
 package org.terasology.entitySystem.systems;
 
 /**
+ * Interface for component systems that needs to be updated every time certain game elements are rendered (generally to render something).
  */
 public interface RenderSystem extends ComponentSystem {
 
+    /**
+     * Called when opaque objects should be rendered.
+     */
     void renderOpaque();
 
+    /**
+     * Called when alpha blended objects should be rendered.
+     */
     void renderAlphaBlend();
 
+    /**
+     * Called when overlay objects should be rendered.
+     */
     void renderOverlay();
 
+    /**
+     * Called when first person objects should be rendered.
+     */
     void renderFirstPerson();
```

Comment:
Leave this as it is: we'll get rid of this method at some point. Everything should be placed in-world. If it's not in world it's either background, overlay, or on the very top UI. Which potentially could be an UI anyway. But I'm thinking out loud here.